sources/packages/api/src/route-controllers/restriction/restriction.controller.ts

```diff
@@ -233,4 +238,185 @@ export class RestrictionController extends BaseController {
       );
     }
   }
+
+  /**
+   * Rest API to get restrictions for an institution.
+   * @param institutionId
+   * @returns Institution Restrictions.
+   */
+  @Groups(UserGroups.AESTUser)
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Get("/institution/:institutionId")
+  async getInstitutionRestrictions(
+    @Param("institutionId") institutionId: number,
+  ): Promise<RestrictionSummaryDTO[]> {
+    const institutionRestrictions =
+      await this.institutionRestrictionService.getInstitutionRestrictionsById(
+        institutionId,
+      );
+    return institutionRestrictions?.map((institutionRestriction) => ({
+      restrictionId: institutionRestriction.id,
+      restrictionType: institutionRestriction.restriction.restrictionType,
+      restrictionCategory:
+        institutionRestriction.restriction.restrictionCategory,
+      description: institutionRestriction.restriction.description,
+      createdAt: institutionRestriction.createdAt,
+      updatedAt: institutionRestriction.updatedAt,
+      isActive: institutionRestriction.isActive,
+    }));
+  }
+
+  /**
+   * Rest API to get the details for view institution restriction.
+   * @param institutionId
+   * @param institutionRestrictionId
+   * @returns Institution restriction detail view.
+   */
+  @Groups(UserGroups.AESTUser)
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Get(
+    "/institution/:institutionId/institutionRestriction/:institutionRestrictionId",
+  )
+  async getInstitutionRestrictionDetail(
+    @Param("institutionId") institutionId: number,
+    @Param("institutionRestrictionId") institutionRestrictionId: number,
+  ): Promise<RestrictionDetailDTO> {
+    const institutionRestriction =
+      await this.institutionRestrictionService.getInstitutionRestrictionDetailsById(
+        institutionId,
+        institutionRestrictionId,
+      );
+    if (!institutionRestriction) {
+      throw new NotFoundException(
+        "The institution restriction does not exist.",
+      );
+    }
+    return {
+      restrictionId: institutionRestriction.id,
+      restrictionType: institutionRestriction.restriction.restrictionType,
+      restrictionCategory:
+        institutionRestriction.restriction.restrictionCategory,
+      description: institutionRestriction.restriction.description,
+      createdAt: institutionRestriction.createdAt,
+      updatedAt: institutionRestriction.updatedAt,
+      createdBy: institutionRestriction.creator
+        ? `${institutionRestriction.creator.lastName}, ${institutionRestriction.creator.firstName}`
+        : "",
+      updatedBy: institutionRestriction.modifier
+        ? `${institutionRestriction.modifier.lastName}, ${institutionRestriction.modifier.firstName}`
```

Comment:
we do have a utils for getting the full name, use can use it, 
![image](https://user-images.githubusercontent.com/77353155/148860118-c446f71a-0ad9-4487-be7f-1eae69cbec56.png)

Reply:
From the Figma, the IDIR user is represented in pattern Lastname, Firstname in Notes and Restrictions. I can write a new util and use it as this pattern used in many places.